obspy/io/cmtsolution/tests/test_core.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+import inspect
+import io
+import os
+import unittest
+
+import obspy
+from obspy.core.util.base import NamedTemporaryFile
+from obspy.io.cmtsolution.core import _is_cmtsolution
+
+
+class CmtsolutionTestCase(unittest.TestCase):
+    """
+    Test suite for obspy.io.cmtsolution.
+
+    The tests usually directly utilize the registered function with the
+    read_events() to also test the integration.
+    """
+    def setUp(self):
+        self.path = os.path.dirname(os.path.abspath(inspect.getfile(
+            inspect.currentframe())))
+        self.datapath = os.path.join(self.path, "data")
+
+    def test_read_and_write_cmtsolution_from_files(self):
+        """
+        Tests that reading and writing CMTSOLUTIONS file does not change
+        anything.
+        """
+        filename = os.path.join(self.datapath, "CMTSOLUTION")
+        with open(filename, "rb") as fh:
+            data = fh.read()
+
+        cat = obspy.read_events(filename)
+
+        with NamedTemporaryFile() as tf:
+            temp_filename = tf.name
+
+        try:
+            cat.write(temp_filename, format="CMTSOLUTION")
+            with open(temp_filename, "rb") as fh:
+                new_data = fh.read()
+        finally:
+            try:
+                os.remove(temp_filename)
+            except:
+                pass
+
+        for line1, line2 in zip(data.decode().splitlines(),
+                                new_data.decode().splitlines()):
+            self.assertEqual(line1, line2)
```

Comment:
I mainly did it to get nice error messages, but turns out the error messages of just passing the lists are also very nice...